src/types.jl

```diff
@@ -21,172 +46,436 @@ abstract type VectorListSxp <: VectorSxp end
+"""
+struct BuiltinSxp <: FunctionSxp
     head::SxpHead
 end
 
-"R character string"
-struct CharSxp <: VectorAtomicSxp     # type tag 9
+"""
+    CharSxp <: VectorAtomicSxp
+
+Representation of R character string.
+
+This corresponds to type tag 9.
+
+# Fields
+- `head::SxpHead`
+- `length::Cint`
+- `truelength::Cint`
+"""
+struct CharSxp <: VectorAtomicSxp
     head::SxpHead
     length::Cint
     truelength::Cint
 end
 
-"R symbol"
-struct SymSxp <: Sxp   # type tag 1
+"""
+    SymSxp <: Sxp
+
+Representation of R symbol.
+
+This corresponds to type tag 1.
+
+# Fields
+- `head::SxpHead`
+- `name::Ptr{CharSxp}`
+- `value::Ptr{UnknownSxp}`
+- `internal::Ptr{UnknownSxp}`
+"""
+struct SymSxp <: Sxp
     head::SxpHead
     name::Ptr{CharSxp}
     value::Ptr{UnknownSxp}
     internal::Ptr{UnknownSxp}
 end
 
-"R logical vector"
-struct LglSxp <: VectorNumericSxp     # type tag 10
+"""
+    LglSxp <: VectorNumericSxp
+
+Representation of R logical vector.
+
+This corresponds to type tag 10.
+
+# Fields
+- `head::SxpHead`
+- `length::Cint`
+- `truelength::Cint`
+"""
+struct LglSxp <: VectorNumericSxp
     head::SxpHead
     length::Cint
     truelength::Cint
 end
 
-"R integer vector"
-struct IntSxp <: VectorNumericSxp     # type tag 13
+""""
+    IntSxp <: VectorNumericSxp
+
+Representation of R integer vector.
+
+This corresponds to type tag 13.
+
+# Fields
+- `head::SxpHead`
+- `length::Cint`
+- `truelength::Cint`
+"""
+struct IntSxp <: VectorNumericSxp
     head::SxpHead
     length::Cint
     truelength::Cint
 end
 
-"R real vector"
-struct RealSxp <: VectorNumericSxp    # type tag 14
+"""
+    RealSxp <: VectorNumericSxp
+
+Representation of R real (numeric) vector.
+
+This correponds to type tag 14.
+
+# Fields
+- `head::SxpHead`
+- `length::Cint`
+- `truelength::Cint`
+"""
+struct RealSxp <: VectorNumericSxp
     head::SxpHead
     length::Cint
     truelength::Cint
 end
 
-"R complex vector"
-struct CplxSxp <: VectorNumericSxp    # type tag 15
+"""
+    CplxSxp <: VectorNumericSxp
+
+Representation of R complex vector.
+
+This corresponds to type tag 15.
+
+# Fields
+- `head::SxpHead`
+- `length::Cint`
+- `truelength::Cint`
+"""
+struct CplxSxp <: VectorNumericSxp
     head::SxpHead
     length::Cint
     truelength::Cint
 end
 
-"R vector of character strings"
-struct StrSxp <: VectorListSxp     # type tag 16
+"""
+    StrSxp <: VectorListSxp
+
+Representation of R vector of character strings.
+
+This correponds to type tag 16.
+
+# Fields
+- `head::SxpHead`
+- `length::Cint`
+- `truelength::Cint`
+"""
+struct StrSxp <: VectorListSxp
     head::SxpHead
     length::Cint
     truelength::Cint
 end
 
-"R dot-dot-dot object"
-struct DotSxp <: Sxp     # type tag 17
+"""
+    DotSxp <: Sxp
+
+Representation of R dot-dot-dot object.
+
+This corresponds to type tag 17.
+
+# Fields
+- `head::SxpHead`
+"""
+struct DotSxp <: Sxp
     head::SxpHead
 end
 
-"R \"any\" object"
-struct AnySxp <: Sxp     # type tag 18
+"""
+    DotSxp <: Sxp
```

Comment:
Should be `AnySxp`, right?

Reply:
```suggestion
    AnySxp <: Sxp
```

Reply:
ah yes, the joy of copy-paste when doing lots of docstrings..